packages/editor/src/extensions/code-block/code-block.ts

```diff
@@ -463,6 +464,17 @@ export const CodeBlock = Node.create<CodeBlockOptions>({
               event.clipboardData
             );
 
+            if (language) {
+              const languageDefinition = Languages.find(
+                (l) =>
+                  l.filename === language ||
+                  l.alias?.some((a) => a === language)
+              );
+              if (languageDefinition) {
+                config.set("codeBlockLanguage", languageDefinition);
+              }
```

Comment:
I don't think we should change the saved code block language when a user pastes some code. It should only be done when the user intentionally selects a different language from the language selector.